### ISSUE TITLE
exponentiation reverts on in-bound result

### ISSUE THREAD

**Issue by charles-cooper** [vyperlang/vyper#2987]
### Version Information

* vyper Version (output of `vyper --version`): https://github.com/vyperlang/vyper/commit/be2c59a604070c75212dca90beb4c33fed908c8b
* Python Version (output of `python --version`): 3.8.10

### What's your issue about?
`-8**5` results in MIN_INT16, but the following code reverts at runtime
```vyper
@external
def foo() -> int16:
    x: int16 = -8
    y: int16 = x ** 5
    return y
```

**Comment by charles-cooper** [comment]
fixed in #2983

**Comment by charles-cooper** [comment]
see https://github.com/vyperlang/vyper/pull/2983/files#diff-08afe8356161d17a86aba4e8afda160c9cc61eed014309071c76b59884afe9dbR77-R87 and impl edge case https://github.com/vyperlang/vyper/pull/2983/files#diff-4e9c9c70ad3c3524c89fb1ae5fed5bb7a379d579619da1000802fbd03373ba74R139-R141
